components/org.wso2.openbanking.cds.consent.extensions/src/main/java/org/wso2/openbanking/cds/consent/extensions/admin/impl/CDSConsentAdminHandler.java

```diff
@@ -296,12 +299,15 @@ private void revokeConsentAsPrimaryUser(DetailedConsentResource detailedConsentR
         String consentID = detailedConsentResource.getConsentID();
         this.consentCoreService.revokeConsentWithReason(consentID, CONSENT_STATUS_REVOKED, null,
                 ConsentCoreServiceConstants.CONSENT_REVOKE_FROM_DASHBOARD_REASON);
-        // revoke access tokens
-        try {
-            consentCoreService.revokeTokens(detailedConsentResource, userId);
-        } catch (IdentityOAuth2Exception e) {
-            log.error(String.format("Error occurred while revoking tokens. Only the consent was revoked " +
-                    "successfully. %s", e.getMessage()));
+        // revoke access tokens if the user is not null. User can be null if the consent is revoked by a user with
+        // CustomerCareOfficer role in consent manager.
+        if (userId != null) {
```

Comment:
Where are we revoking the tokens in CustomerCareOfficer scenario?

Reply:
The `revokeConsentWithReason` [1] method calls > `revokeConsentWithReason` [2] in accelerator which calls > `revokeTokens` [3] method.


[1] https://github.com/wso2/reference-implementation-consumerdatastandards-au/blob/8dfced3424d93cdc11c0ce0c0f535aaaed5d763d/components/org.wso2.openbanking.cds.consent.extensions/src/main/java/org/wso2/openbanking/cds/consent/extensions/admin/impl/CDSConsentAdminHandler.java#L300

[2] https://github.com/wso2/financial-services-accelerator/blob/f1f8302b549aefa6a1ecf403de2ba19ddbf77b72/open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.mgt.service/src/main/java/com/wso2/openbanking/accelerator/consent/mgt/service/impl/ConsentCoreServiceImpl.java#L302

[3] https://github.com/wso2/financial-services-accelerator/blob/f1f8302b549aefa6a1ecf403de2ba19ddbf77b72/open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.mgt.service/src/main/java/com/wso2/openbanking/accelerator/consent/mgt/service/impl/ConsentCoreServiceImpl.java#L372